components/servo/site_data_manager.rs

```diff
@@ -2,22 +2,97 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
 
+use bitflags::bitflags;
 use net_traits::ResourceThreads;
+use rustc_hash::FxHashMap;
+use storage_traits::StorageThreads;
+use storage_traits::webstorage_thread::{OriginDescriptor, StorageType};
+
+bitflags! {
+    #[derive(Clone, Copy, Debug, PartialEq)]
+    pub struct StorageTypes: u32 {
+        const LOCAL_STORAGE   = 1 << 0;
+        const SESSION_STORAGE = 1 << 1;
```

Comment:
Do you mind adding some rustdoc, perhaps with links to the API either in MDN or the specification to explain what these values mean?

```suggestion
        const Local   = 1 << 0;
        const Session = 1 << 1;
```